### ISSUE TITLE
Block bindings: Fix accessibility of the 'Connected' visual indicator in the block toolbar 

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#60810]
### Description

Spltting this out from https://github.comsically/WordPress/gutenberg/issues/58595#issuecomment-2060646354

The 'connected' visual indicator added to the block toolbar is not OK. It's [a focusable div element](https://github.com/WordPress/gutenberg/pull/59185/files#diff-53c7becd18a8bab09d6e368f39a58eacb5ac5dcb25a80b2826308dedeaafc570R12) with an aria-label attribute, which is not OK and it's something that was already pointed out several times as a bad practice. Never use a focusable div unless it reproduces _all_ the native semantics and behaviors of the corresponding HTML element that is supposed to replace.

Additionally:
- It doesn't show a focus style, although it is focusable.
- It doesn't show a Tooltip.

It should be changed to an aria-disabled, nooped, button. It should show a tooltip.

As mentioned in several other issues and PRs, please remind:

> [!IMPORTANT]
> Never ever use focusable div elements.

### Step-by-step reproduction instructions

- Use the code snippets provided at https://github.com/WordPress/gutenberg/issues/58595#issuecomment-2060853669 to have a few bound blocks in your post.
- Select one of the bound Paragraphs.
- Use a keyboard and press Shift + Tab to move focus to the block toolbar.
- Use the left and arrow keys to move through the block toolbar buttons.
- Observe the 'Connected' indicator receives focus but it doesn't show a focus style.
- Observe the 'Connected' indicator doesn't show a tooltip.

Animated GIF to illustrate:

![connected div button](https://github.com/WordPress/gutenberg/assets/1682452/5c6a8cb3-6d3b-4ae8-bd02-64605ed56a01)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by SantosGuillamot** [comment]
The "Connected" icon is not used anymore and instead, the block icon is reused adding a message saying "Block is connected". I'm pretty sure that there are more accessibility improvements we need to work on, but should we close this specific issue as it doesn't seem to apply anymore?

![Screenshot 2024-06-27 at 13 24 05](https://github.com/WordPress/gutenberg/assets/34552881/e5a02ae5-b2c3-4acd-8756-878ebae8e267)

**Comment by afercia** [comment]
Late to the party but I'm afraid the text `This block is connected` placed at the end fo the menu breaks [the ARIA menu pattern](https://www.w3.org/WAI/ARIA/apg/patterns/menubar/).

A `role=menu` is only expected to contain children with role `menuitem`, `menuitemcheckbox`, `menuitemradio` and, optionally, groups with separators.

The text`This block is connected.` is just a `<span>` element placed within a `role="group"`. A menu group is not expected to contain plain text

Also, the arrows keyboard interaction, which is the expected interaction for a menu, will never reach this text as it works only for items in the menu. Screen reader users will have very few chances to discover this text at the bottom of the menu as they wouldn't expect it to be there.

I haven't followed the discussion where this design was provided but there have been already other cases of extraneous content in a role=menu and they have been removed. I'd kindly ask to remove this text from the menu to preserve the ARIA menu pattern expected semantics.

I'd argue that also from a visuals perspective, burying down in a menu such important information isn't ideal
I'd tend to think it's unlikely users will ever expect the 'connected' information to be inside the 'Transform to'  It can't be added in the block toolbar as well. As an alternative solutoin, I'd sugget to explore placing this information in the inspetor block card.

**Comment by SantosGuillamot** [comment]
Thanks a lot for the feedback and sorry for introducing those issues.

If I am not mistaken, that was introduced in this pull request by @kevin940726 and the discussion happened [here](https://github.com/WordPress/gutenberg/pull/62317#discussion_r1629001333).

@richtabor @jasmussen As you were involved in that conversation, maybe you can help here. Any alternative to the "The block is connected" text, which is causing accessibility issues?

**Comment by jasmussen** [comment]
It sounds like it's the placement in the DOM of the text that's causing issues, not the content of the text. The pattern is similar to this one:

![Screenshot 2024-07-03 at 09 47 06](https://github.com/WordPress/gutenberg/assets/1204802/a74023e0-c5fd-425b-8f27-36e7e8445619)

Should that menu be fixed too?

**Comment by kevin940726** [comment]
> As an alternative solutoin, I'd sugget to explore placing this information in the inspetor block card.

Happy to explore this if we think it'd be better for accessibility. It'd be better to have a unified way to display such text as shown in the edit mode popover too though!

**Comment by afercia** [comment]
> The pattern is similar to this one:

Visually, it's similar. Semantically, it is not. That text is placed _outside_ of the `role=menu`:

![Screenshot 2024-07-04 at 09 55 52](https://github.com/WordPress/gutenberg/assets/1682452/361afb8d-1b28-4fe0-896d-633ed03bcd6a)

It's unlikely screen reader users will ever get that text though, unless they switch their scren reader to browse mode and manually explore the content of the popover.
